nemo/collections/asr/data/audio_to_text_dataset.py

```diff
@@ -398,6 +400,88 @@ def get_tarred_dataset(
     return get_chain_dataset(datasets=datasets, ds_config=config, rank=global_rank)
 
 
+def get_code_switched_dataset(
+    config: dict,
+    shuffle_n: int,
+    global_rank: int,
+    world_size: int,
+    tokenizer: Optional['TokenizerSpec'] = None,
+    augmentor: Optional['AudioAugmentor'] = None,
+) -> CodeSwitchedDataset:
+
+    tarred_audio_filepaths = config.get('tarred_audio_filepaths', None)
+    manifest_filepaths = config['manifest_filepath']
+
+    # needed to support validation Datasets that arrive here as
+    # [[dataset1,dataset2]] otherwise ModelPT would interfere
+    if len(manifest_filepaths) == 1 and not isinstance(manifest_filepaths[0], str):
+        manifest_filepaths = config['manifest_filepath'][0]
+    if tarred_audio_filepaths is None:
+        tarred_audio_filepaths = [None] * len(manifest_filepaths)
+    datasets = {}
+    for dataset_idx, (tarred_audio_filepath, manifest_filepath) in enumerate(
+        zip(tarred_audio_filepaths, manifest_filepaths)
+    ):
+        lang = config['code_switch_languages'][dataset_idx]
+        conf = copy.deepcopy(config)
+        conf['manifest_filepath'] = manifest_filepath
+        with open_dict(conf):
+            conf['tarred_audio_filepaths'] = tarred_audio_filepath
+            conf['language'] = lang
+        if tarred_audio_filepath is None or len(tarred_audio_filepath) == 0:
+            if tokenizer is None:
+                dataset = get_char_dataset(config=conf, augmentor=None)
+            else:
+                dataset = get_bpe_dataset(config=conf, tokenizer=tokenizer, augmentor=None)
+        else:
+            dataset = get_tarred_dataset(
+                config=conf,
+                tokenizer=tokenizer,
+                shuffle_n=shuffle_n,
+                global_rank=global_rank,
+                world_size=world_size,
+                augmentor=None,
+            )
+        datasets[lang] = dataset
+
+    config = OmegaConf.to_container(config)
+
+    if (
+        'code_switch_probabilities' in config
+        and config['code_switch_probabilities'] is not None
+        and len(config['code_switch_probabilities']) == len(config['code_switch_languages'])
+    ):
+        cs_probs_dict = {
+            l: config['code_switch_probabilities'][i] for i, l in enumerate(config['code_switch_languages'])
+        }
+    else:
+        cs_probs_dict = None
+
+    dataset = CodeSwitchedDataset(
```

Comment:
is this a tarred dataset equivalent? Or loose file dataset?

Reply:
If tarred it needs to be marked in its name as such

Reply:
Neither really. It's the same "style" as the ConcatDataset

Reply:
We now have multiple wrapper classes; i think it's time to start thinking about a proper graph of such. Not sure that it's for this PR to cover, but we ought to consider, i think.